### PR TITLE
Remove JasperDbSettingsImpl and move everything to JasperDbConfig

### LINKED ISSUE #6379: Remove `JasperDbSettingsImpl` and move everything to `JasperDbConfig`
Refactor every access from `Settings.getJasperDb()` to use new configuration API

### PR DESCRIPTION
**Pull request by timo0**
Fix #6379

Remove JasperDbSettingsImpl and move everything to JasperDbConfig

Removed:
* DefaultJasperDbSettings
* JasperDbSettings
* JasperDbSettingsFactory
* JasperDbSettingsImpl
* JasperDbSettingsImplTest
* TestJasperDbSettings

### CODE REVIEW

**Review by edward-swirldslabs — APPROVED** [review]

**Review by hendrikebbers — APPROVED** [review]

**Review by imalygin — APPROVED** [review]

**Review by edward-swirldslabs — DISMISSED** [review]

**Review by imalygin — DISMISSED** [review]

File: platform-sdk/swirlds-merkle/src/test/java/com/swirlds/virtual/merkle/reconnect/BreakableDataSourceJPDB.java [hunk 1]
```diff
@@ -34,7 +34,12 @@ public BreakableDataSourceJPDB(final BrokenBuilder builder, final VirtualDataSou
 
     @Override
     public VirtualKeySet<TestKey> buildKeySet() {
-        return new HalfDiskVirtualKeySet<>(
-                new TestKeySerializer(), 10, 2L * MEBIBYTES_TO_BYTES * BYTES_TO_BITS, 1_000_000, 10_000);
+        final Configuration configuration = new TestConfigBuilder()
+                .withValue("keySetHalfDiskHashMapSize", "10000")
```
Comment (hendrikebbers) [inline platform-sdk/swirlds-merkle/src/test/java/com/swirlds/virtual/merkle/reconnect/BreakableDataSourceJPDB.java:38]: sicher, dass da nicht ein prefix fehlt für den Namen?

File: platform-sdk/swirlds-merkle/src/test/java/com/swirlds/virtual/merkle/reconnect/VirtualMapJPDBReconnectTest.java [hunk 2]
```diff
@@ -61,37 +60,24 @@
...
-                return 10;
-            }
+        originalConfig = ConfigurationHolder.getInstance().get();
 
-            @Override
-            public long getKeySetBloomFilterSizeInBytes() {
-                return 2 * MEBIBYTES_TO_BYTES * BYTES_TO_BITS;
-            }
+        final Configuration config = new TestConfigBuilder()
+                .withValue("keySetHalfDiskHashMapSize", "10000")
```
Comment (hendrikebbers) [inline platform-sdk/swirlds-merkle/src/test/java/com/swirlds/virtual/merkle/reconnect/VirtualMapJPDBReconnectTest.java:70]: prefix?

File: platform-sdk/swirlds-platform-core/src/test/java/com/swirlds/platform/reconnect/RandomVirtualMapJPDBReconnectTests.java [hunk 3]
```diff
@@ -66,37 +65,24 @@ class RandomVirtualMapJPDBReconnectTests extends VirtualMapReconnectTestBase {
...
-                return 10;
-            }
+        originalConfig = ConfigurationHolder.getInstance().get();
 
-            @Override
-            public long getKeySetBloomFilterSizeInBytes() {
-                return 2 * MEBIBYTES_TO_BYTES * BYTES_TO_BITS;
-            }
+        final Configuration config = new TestConfigBuilder()
+                .withValue("keySetHalfDiskHashMapSize", "10000")
```
Comment (hendrikebbers) [inline platform-sdk/swirlds-platform-core/src/test/java/com/swirlds/platform/reconnect/RandomVirtualMapJPDBReconnectTests.java:75]: prefix?

File: platform-sdk/swirlds-merkle/src/test/java/com/swirlds/virtual/merkle/reconnect/VirtualMapJPDBReconnectTest.java (hunk 2 above)
Comment (timo0) [reply platform-sdk/swirlds-merkle/src/test/java/com/swirlds/virtual/merkle/reconnect/VirtualMapJPDBReconnectTest.java]: thanks

**Review by imalygin — APPROVED** [review]

**Review by nathanklick — DISMISSED** [review]
Approval limited to the `module-info.java` changes only. Please seek other reviews.

**Review by hendrikebbers — DISMISSED** [review]

**Review by edward-swirldslabs — APPROVED** [review]

File: platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/jasperdb/config/JasperDbConfig.java [hunk 4]
```diff
@@ -134,7 +134,8 @@ public record JasperDbConfig(
         @ConfigProperty(defaultValue = "2147483648") long keySetBloomFilterSizeInBytes,
         @ConfigProperty(defaultValue = "1000000000") long keySetHalfDiskHashMapSize,
         @ConfigProperty(defaultValue = "1000000") int keySetHalfDiskHashMapBuffer,
-        @ConfigProperty(defaultValue = "false") boolean indexRebuildingEnforced) {
+        @ConfigProperty(defaultValue = "false") boolean indexRebuildingEnforced,
+        @ConfigProperty(defaultValue = "1048576") int leafRecordCacheSize) {
```
Comment (hendrikebbers) [inline platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/jasperdb/config/JasperDbConfig.java:138]: do we have documentation for the new property?

**Review by imalygin — APPROVED** [review]

**Review by edward-swirldslabs — DISMISSED** [review]

**Review by edward-swirldslabs — APPROVED** [review]

**Review by hendrikebbers — APPROVED** [review]

**Review by imalygin — APPROVED** [review]

**Pull request merged**